core/src/Kokkos_View.hpp

```diff
@@ -542,7 +552,7 @@ class View : public ViewTraits< DataType , Properties ... > {
 
 private:
 
-  typedef Kokkos::Impl::ViewMapping< traits , void > map_type ;
+  typedef Kokkos::Impl::ViewMapping< traits , typename traits::specialize > map_type ;
```

Comment:
@crtrott I think this change will affect Sacado, the `ViewMapping` there expects `void`, but this change will result (in Sacado) in the second template parameter being some specialize tag (e.g. `ViewSpecializeSacadoFad` etc). The change there seems like it should be easy, but we should check with Eric in case of overlooking something.

Reply:
Yeah, look at my new commit which fixes the internal aggregates test which mimics what Sacado does. It actually makes the code cleaner and I was able to delete parts of the specialization.

Reply:
The only potential issue I see is that Sacado currently has several partial specializations of ViewMapping<> for a given view specialization tag that are differentiated through enable_if<>, to support assignments between different view specializations (see, e.g., the ViewMapping<> specializations [here](https://github.com/trilinos/Trilinos/blob/master/packages/sacado/src/KokkosExp_View_Fad_Contiguous.hpp).)  This is to allow, e.g, View<double...> = View<Fad...> etc.  Will that still work?

Reply:
@etphipp it may be as simple as replacing the final explicit `void` template parameter in `ViewMapping` typedefs with the appropriate specialization tag except for the case you mentioned above, e.g. leave as `void` for the `DstType` in this case:
[line 1365](https://github.com/trilinos/Trilinos/blob/cdbfaf56f44997486e915d967f664d5108070ebe/packages/sacado/src/KokkosExp_View_Fad_Contiguous.hpp#L1365-L1366) and in matching places for the other specialized `ViewMapping` types.

Also making sure to change the default void return type of the `enable_if`s used in the `ViewMapping` specialization declarations to return the proper specialize tag.